### ISSUE TITLE
No autocompletion despite static typing

### ISSUE THREAD

**Issue by CappedMonke** [godotengine/godot#73638]
### Godot version

Godot 4.0 RC 2

### System information

Windows 11

### Issue description

As the title states you get no autocompletion when statically typing.
Of course the code still works, but it shouldn't be neccessary to cast an object to a type it already is.

### Steps to reproduce

1. Create child class with some functionality
![image](https://user-images.githubusercontent.com/100119945/220173809-0b2b2788-d986-4fa7-815c-444268d09c32.png)

2. Get node in parent class and call childs function
![image](https://user-images.githubusercontent.com/100119945/220174167-5ff3c8e9-863e-4fbd-a3cf-d88d34597b39.png)

### Minimal reproduction project

N/A

**Comment by poohcom1** [comment]
I think this specific to auto-completion for variables obtained through `get_node`. There's a feature where the editor will provide auto-completion for relative node based on the currently opened scene, but it seems like it's incorrectly overriding the static typing in this case:

| Without `get_node` | With `get_node` |
| -- | -- |
| ![image](https://user-images.githubusercontent.com/74857873/220936230-19f2109e-58e4-4403-97c6-e7241d5c2329.png) | ![image](https://user-images.githubusercontent.com/74857873/220936294-8b4cbb2f-6380-4b8e-a29f-6b9be5033a05.png) |

I still get auto-completion thought, just not for `ChildClass`, but for `Node2D` since that's the type of the node being referenced. Is this also the case for you, or are you just not getting any auto-completion at all?

If so, then the main issue here is that auto-completion from `get_node` has higher precedence than static typing auto-completion, which shouldn't be the case. Ideally static typing should take precedence, or the results should be combined.

**Comment by CappedMonke** [comment]
Yes, you still get the Node2D auto-completion, exactly as you said.

**Comment by sam-s4s** [comment]
I came here to add this issue, but luckily found this first. Seems quite important to me, as a newcomer to Godot.

The first thing I tried to do was type some things in the tutorial, and it failed straight away.

```
var mob_spawn_location: PathFollow2D = get_node("MobPath/MobSpawnLocation")
mob_spawn_location.progre # autocomplete should give you .progress_ratio here, but it does not
```

Current workaround is to override the type from get_node, as described above...

```
var mob_spawn_location := get_node("MobPath/MobSpawnLocation") as PathFollow2D
```

But it is confusing and catches newbies like me out :P

**Comment by strangebirds** [comment]
I can confirm also present in Mac (Monterey) on Godot 4.01

**Comment by 0Volcanon** [comment]
Just ran into this problem on Godot 4.0.2 while writing some tool scripts. The problem only seems to exist for the variables defined outside of the functions. If the hint is provided inside the function, it works fine. This problem did not exist in Godot 3.5.2.

**Comment by sam-s4s** [comment]
> Just ran into this problem on Godot 4.0.2 while writing some tool scripts. The problem only seems to exist for the variables defined outside of the functions. If the hint is provided inside the function, it works fine. This problem did not exist in Godot 3.5.2.

My example above is all happening inside a function `func _on_MobTimer_timeout():`

(as per the https://docs.godotengine.org/en/3.5/getting_started/first_3d_game/05.spawning_mobs.html tutorial)

**Comment by 0Volcanon** [comment]
I checked and it existed in 3.5 and 3.5.1 and was fixed in 3.5.2. #48037

**Comment by mbrlabs** [comment]
In addition to @poohcom1's comment:
> I think this is specific to auto-completion for variables obtained through `get_node`

If you use `get_node()` or the short hand version you can get autocompletion in the script that is referencing it, but only if the scene for `ChildClass` is currently open.
